### PR TITLE
Some improvements 

### PR DESCRIPTION
**Pull request by giannivh**
- In my (default) Mac OS X setup highlighted messages didn't came through, fixed by casting highlight -> int(highlight)
- Clearer titles
- New option to show all other messages with a different sound setting
- Don't show notification if message comes from own nickname

### CODE REVIEW

**Comment by sindresorhus** [comment]
I'm happy to include the first and last point, but not interested in the rest.

File: readme.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 > [WeeChat](http://www.weechat.org) script to pass highlights and private messages to the OS X 10.8+ Notification Center.
 
+Forked and adapted:
+- Option to show all messages
+- Updated titles to be more clear
```
Comment (sindresorhus) [inline readme.md]: I guess you forgot to remove this.

File: notification_center.py [hunk 2]
```diff
@@ -17,22 +22,35 @@
...
 def notify(data, buffer, date, tags, displayed, highlight, prefix, message):
-	# passing `None` or `''` still plays the default sound so we pass a lambda instead
+
+	#If own nickname is sender, ignore
+	mynick = weechat.buffer_get_string(buffer,'localvar_nick')
+	if prefix == mynick or prefix == ('@%s' % mynick):
+		return weechat.WEECHAT_RC_OK
+
+	#Format message
+	message = repr(message.encode('string-escape'))
```
Comment (sindresorhus) [inline notification_center.py]: Why?

File: notification_center.py [hunk 3]
```diff
@@ -17,22 +22,35 @@
...
+	'show_other_message': 'on',
+	'sound': 'on',
+	'sound_on_other_message': 'off'
 }
 
 for key, val in DEFAULT_OPTIONS.items():
 	if not weechat.config_is_set_plugin(key):
 		weechat.config_set_plugin(key, val)
 
-weechat.hook_print('', 'irc_privmsg', '', 1, 'notify', '')
```
Comment (sindresorhus) [inline notification_center.py]: Don't no unrelated changes.

File: notification_center.py (hunk 2 above)
Comment (giannivh) [inline notification_center.py]: Because if you want a notification for "other messages" it also shows the messages you send.

The format was because the messages I received contained code which messed up pync. You can ignore it.

**Comment by giannivh** [comment]
Feel free to take what you want.

**Comment by sindresorhus** [comment]
> In my (default) Mac OS X setup highlighted messages didn't came through, fixed by casting highlight -> int(highlight)

I'd like to figure out why before I try to fix it. What value did it return for you?

**Comment by giannivh** [comment]
```python
if highlight == '1':
```
was always false in my case: not a string.

**Comment by sindresorhus** [comment]
@giannivh I don't use Weechat anymore. Would you be able to do me a favor and test `master` and make sure it works before I submit a new version?

**Comment by limpep** [comment]
Any chance we can get this approved and merged?

**Comment by sindresorhus** [comment]
@limpep This is not getting merged. I've cherrypicked what I want in [master](https://github.com/sindresorhus/weechat-notification-center). If you can confirm it's working I will do a release ;)

**Comment by limpep** [comment]
@sindresorhus The only part that is working on 10.9.5 with ITerm2 is when someone sends you a private message, highlights are not working.

**Comment by sindresorhus** [comment]
I would be happy to do a new release when someone confirms/fixes master. I don't have time or interest in setting up WeeChat again just to test it.

https://github.com/sindresorhus/weechat-notification-center/issues/6
